src/main/java/persistence/entity/EntityPersister.java

```diff
@@ -0,0 +1,40 @@
+package persistence.entity;
+
+import jdbc.JdbcTemplate;
+import persistence.sql.dml.DeleteQuery;
+import persistence.sql.dml.FindByIdQuery;
+import persistence.sql.dml.GenericRowMapper;
+import persistence.sql.dml.InsertQuery;
+import persistence.sql.dml.UpdateQuery;
+
+public class EntityPersister<T> {
+
+    private final Class<T> entityClass;
+    private final JdbcTemplate jdbcTemplate;
+
+    public EntityPersister(Class<T> entityClass, JdbcTemplate jdbcTemplate) {
+        this.entityClass = entityClass;
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public T findById(Object primaryKey) {
+        String sql = new FindByIdQuery<>(entityClass, primaryKey).generateQuery();
```

Comment:
```suggestion
        String sql = new FindByIdQuery<>(entityClass).generateQuery(primaryKey);
```
변하는 값을 내부 상태에서 제거하면, 객체를 재활용이 가능할지도 모르겠네요 🤔

Reply:
FindQuery 클래스로 변경하여 id로 조회하는 쿼리와 findAll 같은 조회 메서드들을 분리해서 사용하면, 객체의 재사용성을 높일 수 있을 것 같습니다. 물론 객체의 책임이 늘어나긴 하겠지만, 재사용성을 고려한다면 좋은 선택이 될 수 있다고 생각합니다.

혹시 제가 놓친 부분이나 잘못된 점이 있다면, 피드백 주시면 감사하겠습니다.